python/tank/hook.py

```diff
@@ -611,14 +611,31 @@ def execute_hook_method(hook_paths, parent, method_name, **kwargs):
     return ret_val
 
 
-def create_hook_instance(hook_paths, parent):
+def create_hook_instance(hook_paths, parent, base_class=None):
```

Comment:
Worth carving a path to passing in a base class for execut_hook_method as well?

Reply:
good call. added.